graph.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
+# graph.py
+from collections import defaultdict
+
+
+class Graph():
+    def __init__(self, vertices):
+        self.graph = defaultdict(list)
+        self.V = vertices
+
+    def add_edge(self, u, v):
+        self.graph[u].append(v)
+
+    # internal
+    def _is_cyclic_util(self, v, visits_p, n):
```

Comment:
I don't think the cycle-finding method has to be so complicated. Each word is a dependent of only one other word, i.e. each node in your graph has only one edge that goes out from it, so the algorithm can be *greatly* simplified. (I think you can figure out the algorithm, but if you need help, feel free to ask! :smiley:)